uat/custom-components/client-java-paho/src/main/java/com/aws/greengrass/testing/mqtt5/client/grpc/GRPCControlServer.java

```diff
@@ -171,7 +171,8 @@ public void createMqttConnection(MqttConnectRequest request,
                     .keepalive(keepalive)
                     .cleanSession(request.getCleanSession())
                     .mqtt50(version == MqttProtoVersion.MQTT_PROTOCOL_V_50)
-                    .connectionTimeout(timeout);
+                    .connectionTimeout(timeout)
+                    .userProperties(request.getProperties().getUserPropertiesMap());
```

Comment:
in proto3 by default all fields are optional. That means  you need check first is properties are present before request.getProperties()